File: backend/app/api/v1/recommendations.py
"""'What to learn next' recommendation endpoints, driven by graph analysis (Phase 7)."""

from fastapi import APIRouter, Depends, Query

from app.api.deps import get_current_user_id
from app.models.recommendation import RecommendationResponse
from app.services import recommendation_service

router = APIRouter(prefix="/recommendations", tags=["recommendations"])


@router.get("", response_model=RecommendationResponse)
async def get_recommendations(
    limit: int = Query(default=recommendation_service.DEFAULT_LIMIT, ge=1, le=50),
    current_user_id: str = Depends(get_current_user_id),
) -> RecommendationResponse:
    """
    Deterministically ranks "gap" (structurally load-bearing but shallowly
    covered) and "next step" (one `prerequisite_of` hop from something
    already well covered) concepts from this user's knowledge graph -- see
    app/services/graph/recommendations.py. No LLM call. Empty
    `concepts`/`edges` (nothing extracted yet) returns `{"items": []}`, not
    an error.
    """
    return await recommendation_service.get_recommendations(current_user_id, limit=limit)

File: frontend/src/lib/api/recommendations.ts
import { apiFetch } from "./client";
import type { RecommendationResponse } from "@/lib/types/recommendation";

/**
 * Client for `/api/v1/recommendations` on the FastAPI backend -- ranked
 * "what to learn next" suggestions, deterministically derived from the
 * knowledge graph on every call (no caching). See
 * backend/app/api/v1/recommendations.py and
 * backend/app/services/recommendation_service.py.
 */
export async function getRecommendations(limit?: number): Promise<RecommendationResponse> {
  const query = typeof limit === "number" ? `?limit=${limit}` : "";
  return apiFetch<RecommendationResponse>(`/api/v1/recommendations${query}`);
}
